File: ionic/ionicTestApp/src/domain/repository/TrainingRepository.ts
import { PrijavljenTrening } from '../models/Trening';
import { Teretana } from '../models/Trening';
import { DostupniTrening } from '../models/Trening';
import { TrainingDetails } from '../models/Trening';
import { SignUpResult } from '../models/common';
import { AttendanceUpdate, AttendanceUpdateResult, CreateRasporedInput, CreateTreningInput, Dvorana, PrijavljeniSudionik, TrainingDropdownItem, TrenerTrening, VrstaTreningaDomain } from '../models/trainer';

export interface TrainingRepository {
  getTrainingsForUser(userId: string): Promise<PrijavljenTrening[]>;

  getAllTeretane(): Promise<Teretana[]>;
  getAvailableTrainings(
    teretanaId: string,
    date: Date
  ): Promise<DostupniTrening[]>;

  getTrainingDetails(treningId: string, rasporedId: String): Promise<TrainingDetails>;

  signUpForTraining(
    rasporedId: string,
    userId: string
  ): Promise<SignUpResult>;

  getTrainingsForTrainer(trenerId: string): Promise<TrenerTrening[]>;
  getAttendeesForRaspored(rasporedId: string): Promise<PrijavljeniSudionik[]>;
  setAttendanceForRaspored(
    rasporedId: string,
    updates: AttendanceUpdate[]
  ): Promise<AttendanceUpdateResult>;

  getDvorane(idTeretane?: string): Promise<Dvorana[]>;
  getVrsteTreninga(): Promise<VrstaTreningaDomain[]>;

  createTrening(input: CreateTreningInput): Promise<void>;
  createRaspored(input: CreateRasporedInput): Promise<void>;
  deleteRaspored(idRasporeda: string): Promise<void>;
  getTrainingsForDropdown(): Promise<TrainingDropdownItem[]>;
}

File: flutter/test_app/lib/domain/repository/training_repository.dart
import '../model/common_models.dart';
import '../model/prijavljen_trening.dart';
import '../model/training_models.dart';
import '../model/trener_models.dart';

abstract class TrainingRepository {
  Future<List<PrijavljenTrening>> getTrainingsForUser(String sportasId);

  Future<List<Teretana>> getTeretane();

  Future<List<DostupniTrening>> getTrainingsByDateAndTeretana({
    required String teretanaId,
    required DateTime date,
  });

  Future<SignUpResult> signUpForTraining({
    required String rasporedId,
    required String korisnikId,
  });

  Future<List<TrainerTraining>> getTrainingsForTrainer(String trenerId);

  Future<List<PrijavljeniSudionik>> getAttendeesForRaspored(String rasporedId);

  Future<AttendanceUpdateResult> setAttendanceForRaspored({
    required String rasporedId,
    required List<AttendanceUpdate> updates,
  });

  Future<List<VrstaTreninga>> getVrsteTreninga();
  Future<List<VrstaTreningaOption>> getVrsteTreningaOptions();

  Future<List<TreningOption>> getTreningOptions();

  Future<void> createTrening(CreateTreningInput input);

  Future<void> createRaspored(CreateRasporedInput input);

  Future<DeleteRasporedResult> deleteRaspored(String rasporedId);

  Future<List<DvoranaSimple>> getDvorane();
  }
